main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1912,6 +1912,13 @@ object Typer {
           resultEff = Type.mkUnion(eff1, eff2, loc)
         } yield (constrs1 ++ constrs2, resultTyp, resultPur, resultEff)
 
+      case KindedAst.Expression.Error(m) =>
```

Comment:
I think it would be better if this had a type variable, purity variable, and effect variable and you unified with those.